src/main/java/com/scalyr/integrations/kafka/AddEventsClient.java

```diff
@@ -265,6 +288,31 @@ private AddEventsResponse parseAddEventsResponse(CloseableHttpResponse httpRespo
     }
   }
 
+  /**
+   * This method takes (potentially) compressed addEventsPayload and if it's compressed, it decompresses it and returns
+   * a decompressed version. It ignores any exceptions which may arise when trying to decompress the payload.
+   *
+   * @param addEventsPayload  byte[] addEvents payload (post compression if compression is enabled).
+   * @return Decompressed payload byte array.
+   */
+  private byte[] getDecompressedPayload(byte[] addEventsPayload) {
+    byte[] decompressedPayload = "unable to decompress the payload".getBytes();
+    InputStream inputStream = compressor.newStreamDecompressor(new ByteArrayInputStream(addEventsPayload));
+
+    try {
+      decompressedPayload = IOUtils.readAllBytes(inputStream);
```

Comment:
It appears we support and use Java 8 on CI so I can't just use much nicer ``try (inputStream) {...`` which should take care of closing the stream, right?

Reply:
Yes, we should use the Java 8 try with resources here.